kernel/aster-nix/src/process/process/mod.rs

```diff
@@ -98,31 +110,57 @@ impl Process {
         sig_dispositions: Arc<Mutex<SigDispositions>>,
         resource_limits: ResourceLimits,
         nice: Nice,
-    ) -> Self {
+    ) -> Arc<Self> {
         let children_pauser = {
             // SIGCHID does not interrupt pauser. Child process will
             // resume paused parent when doing exit.
             let sigmask = SigMask::from(SIGCHLD);
             Pauser::new_with_mask(sigmask)
         };
 
-        Self {
-            pid,
-            threads: Mutex::new(threads),
-            executable_path: RwLock::new(executable_path),
-            process_vm,
-            children_pauser,
-            status: Mutex::new(ProcessStatus::Uninit),
-            parent: Mutex::new(parent),
-            children: Mutex::new(BTreeMap::new()),
-            process_group: Mutex::new(Weak::new()),
-            file_table,
-            fs,
-            umask,
-            sig_dispositions,
-            resource_limits: Mutex::new(resource_limits),
-            nice: Atomic::new(nice),
-        }
+        let process = Arc::new_cyclic(|process_ref: &Weak<Process>| {
+            let current_process = process_ref.clone();
+            let sent_signal = move || {
+                let signal = KernelSignal::new(SIGALRM);
+                if let Some(process) = current_process.upgrade() {
+                    process.enqueue_signal(signal);
+                }
+            };
+
+            let work_sent_signal = {
+                let work_func = Box::new(sent_signal);
+                let work_item = Arc::new(WorkItem::new(work_func));
+                move || {
+                    submit_work_item(
+                        work_item.clone(),
+                        crate::thread::work_queue::WorkPriority::High,
```

Comment:
`work_item` does not need to be cloned

Reply:
Cloning `work_item` here is because `work_sent_signal` is a `Fn` instead of `FnOnce`, so it may be invoked for multiple times by the timer, so it should be cloned each time the closure is be invoked.